Scripts/Navigation/NavigableScrollElement.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+namespace DUCK.DebugMenu.Navigation
+{
+	public class NavigableScrollElement : MonoBehaviour, ISelectHandler
+	{
+		private static NavigableScrollElement tweeningElement;
+
+		[SerializeField]
+		private float scrollSpeed = 1000f;
+
+		private ScrollRect scrollRect;
+		private RectTransform rootCanvasTransform;
+		private RectTransform scrollRectTransform;
+		private float scrollRectTop;
+		private float scrollRectBottom;
+
+		private void Start()
+		{
+			scrollRect = GetComponentInParent<ScrollRect>();
+			rootCanvasTransform = (RectTransform) GetComponentInParent<Canvas>().transform;
+			scrollRectTransform = ((RectTransform) scrollRect.transform);
+
+			// get world space corners of the scroll rect
+			Vector3[] corners = new Vector3[4];
+			scrollRectTransform.GetWorldCorners(corners);
+
+			// now we know the min/max y values of the scroll rect in "canvas space"
+			scrollRectTop = rootCanvasTransform.InverseTransformPoint(corners[1]).y;
+			scrollRectBottom = rootCanvasTransform.InverseTransformPoint(corners[0]).y;
+		}
+
+		public void OnSelect(BaseEventData eventData)
+		{
+			// work out if this element is fully visible to the scroll view
```

Comment:
Looks like this comment should of been removed

Reply:
That's describing what the function is doing

Reply:
The next comment, 2 lines down with no logic in-between starts off with "now get" which suggests something just happened before it. Its rather confusing - maybe put the comment that is "describing what the function is doing" in the method summary